cmd/dagger/env.go

```diff
@@ -0,0 +1,645 @@
+	case envconfig.SDKGo:
+		runCodegenFunc = func() error {
+			if engineClient == nil {
+				var err error
+				engineClient, ctx, err = client.Connect(ctx, client.Params{
+					RunnerHost: engine.RunnerHost(),
+				})
+				if err != nil {
+					return fmt.Errorf("failed to connect to engine: %w", err)
+				}
+			}
+			c, err := dagger.Connect(ctx, dagger.WithConn(EngineConn(engineClient)))
+			if err != nil {
+				return fmt.Errorf("failed to connect to engine: %w", err)
+			}
+			envFlagCfg := &environmentFlagConfig{local: &localEnvironment{path: path}}
+			deps, err := envFlagCfg.loadDeps(ctx, c)
+			if err != nil {
+				return fmt.Errorf("failed to load dependencies: %w", err)
+			}
+			return RunCodegen(ctx, engineClient, nil, newEnvCfg, deps, cmd, nil)
+		}
+	case envconfig.SDKPython:
+	default:
+		return fmt.Errorf("unsupported environment SDK: %s", sdk)
+	}
+
+	cfgBytes, err := json.MarshalIndent(newEnvCfg, "", "  ")
+	if err != nil {
+		return fmt.Errorf("failed to marshal environment config: %w", err)
+	}
+	parentDir := filepath.Dir(path)
+	_, parentDirStatErr := os.Stat(parentDir)
+	switch {
+	case parentDirStatErr == nil:
+		// already exists, nothing to do
+	case os.IsNotExist(parentDirStatErr):
+		// make the parent dir, but if something goes wrong, clean it up in the defer
+		if err := os.MkdirAll(parentDir, 0755); err != nil {
+			return fmt.Errorf("failed to create environment config directory: %w", err)
+		}
+		defer func() {
+			if rerr != nil {
+				os.RemoveAll(parentDir)
+			}
+		}()
+	default:
+		return fmt.Errorf("failed to stat parent directory: %w", parentDirStatErr)
+	}
+
+	var cfgFileMode os.FileMode = 0644
+	originalContents, configFileReadErr := os.ReadFile(path)
+	switch {
+	case configFileReadErr == nil:
+		// attempt to restore the original file if it already existed and something goes wrong
+		stat, err := os.Stat(path)
+		if err != nil {
+			return fmt.Errorf("failed to stat environment config: %w", err)
+		}
+		cfgFileMode = stat.Mode()
+		defer func() {
+			if rerr != nil {
+				os.WriteFile(path, originalContents, cfgFileMode)
+			}
+		}()
+	case os.IsNotExist(configFileReadErr):
+		// remove it if it didn't exist already and something goes wrong
+		defer func() {
+			if rerr != nil {
+				os.RemoveAll(path)
+			}
+		}()
+	default:
+		return fmt.Errorf("failed to read environment config: %w", configFileReadErr)
+	}
+
+	// nolint:gosec
+	if err := os.WriteFile(path, append(cfgBytes, '\n'), cfgFileMode); err != nil {
+		return fmt.Errorf("failed to write environment config: %w", err)
+	}
+
+	if err := runCodegenFunc(); err != nil {
+		return fmt.Errorf("failed to run codegen: %w", err)
+	}
+
+	return nil
+}
+
+func getEnvironmentFlagConfig() (*environmentFlagConfig, error) {
+	environmentURI := environmentURI
+	if environmentURI == "" || environmentURI == environmentURIDefault {
+		// it's unset or default value, use env if present
+		if v, ok := os.LookupEnv("DAGGER_PROJECT"); ok {
```

Comment:
Should probably be renamed:

```suggestion
		if v, ok := os.LookupEnv("DAGGER_ENV"); ok {
```

(feel free to ignore pending our open terminology questions though)